### DISCUSSION TITLE
Does erasure encoding support modifying the default block size?

### DISCUSSION THREAD

**Discussion by FourSpaces** [minio/minio#19126] [Q&A] [answered]
Does erasure encoding support modifying the default block size? Will a 1MB block size cause an increase in IOPS and disk utilization when writing to an HDD disk?

**Comment by harshavardhana** [comment, marked as answer]
EC block is per stream block, each stream itself is read at 2 or 4mb blocks when written to disk which is the optimal block for HDD.

If you want IOPs don't even use HDDs. HDDs are never going to give you IOPs. Deploy NVMe based cluster if you want a IOPs focused cluster.

**Comment by FourSpaces** [comment]
Because I saw in the code that each block is approximately 1M in size, and each block is evenly distributed in an EC set when writing data. If the EC set is relatively large, the data allocated to the disks in each EC set is small files at the KB level. Will this not result in smaller IO being written each time?

**Reply by klauspost** [reply]
No. There are also buffering for stream writes. Also there is considerable downsides with bigger blocks. It cannot and should not be tinkered with.
